### PR TITLE
[SPARK-35847][PYTHON] Manage InternalField in DataTypeOps.isnull

### PR DESCRIPTION
**Pull request by ueshin**
### What changes were proposed in this pull request?

Properly set `InternalField` for `DataTypeOps.isnull`.

### Why are the changes needed?

The result of `DataTypeOps.isnull` must always be non-nullable boolean.
We should manage `InternalField` for this case.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Added some more tests.

### CODE REVIEW

**Comment by ueshin** [comment]
cc @xinrong-databricks @Yikun

File: python/pyspark/pandas/tests/data_type_ops/test_num_ops.py [hunk 1]
```diff
@@ -320,29 +320,55 @@ def test_astype(self):
...
+    @property
+    def intergral_extension_pssers(self):
+        return [ps.from_pandas(pser) for pser in self.intergral_extension_psers]
+
+    @property
+    def intergral_extension_pser_psser_pairs(self):
+        return zip(self.intergral_extension_psers, self.intergral_extension_pssers)
+
+    def test_from_to_pandas(self):
+        for pser, psser in self.intergral_extension_pser_psser_pairs:
```
Comment (ueshin) [inline python/pyspark/pandas/tests/data_type_ops/test_num_ops.py:337]: @xinrong-databricks Is this change fine?

File: python/pyspark/pandas/tests/data_type_ops/test_num_ops.py [hunk 2]
```diff
@@ -320,29 +320,55 @@ def test_astype(self):
...
+    @property
+    def fractional_extension_pssers(self):
+        return [ps.from_pandas(pser) for pser in self.fractional_extension_psers]
+
+    @property
+    def fractional_extension_pser_psser_pairs(self):
+        return zip(self.fractional_extension_psers, self.fractional_extension_pssers)
+
+    def test_from_to_pandas(self):
+        for pser, psser in self.fractional_extension_pser_psser_pairs:
```
Comment (ueshin) [inline python/pyspark/pandas/tests/data_type_ops/test_num_ops.py:364]: ditto.

**Review by HyukjinKwon — APPROVED** [review]

**Review by Yikun — APPROVED** [review]
Thanks for cc me, the DecimalOps and FractionalOps test case are already covered and added the ExtensionOps testcases. 

So, LGTM!

File: python/pyspark/pandas/tests/data_type_ops/test_num_ops.py [hunk 3]
```diff
@@ -320,29 +320,55 @@ def test_astype(self):
...
+        return zip(self.intergral_extension_psers, self.intergral_extension_pssers)
+
+    def test_from_to_pandas(self):
+        for pser, psser in self.intergral_extension_pser_psser_pairs:
             self.check_extension(pser, psser.to_pandas())
             self.check_extension(ps.from_pandas(pser), psser)
 
+    def test_isnull(self):
+        for pser, psser in self.intergral_extension_pser_psser_pairs:
+            self.check_extension(pser.isnull(), psser.isnull())
```
Comment (Yikun) [inline python/pyspark/pandas/tests/data_type_ops/test_num_ops.py:343]:
```python
self.assert_eq(self.pser.isnull(), self.psser.isnull())
```

It should be `assert_eq`, the `psser.isnull()` return value is not ExtensionDtypes.

Comment (ueshin) [reply python/pyspark/pandas/tests/data_type_ops/test_num_ops.py]: Yeah, that's right. Thanks!

File: python/pyspark/pandas/tests/data_type_ops/test_num_ops.py (hunk 1 above)
Comment (xinrong-databricks) [reply python/pyspark/pandas/tests/data_type_ops/test_num_ops.py:337]: Certainly!

File: python/pyspark/pandas/tests/data_type_ops/test_num_ops.py [hunk 4]
```diff
@@ -320,29 +320,55 @@ def test_astype(self):
...
+    @property
+    def intergral_extension_pser_psser_pairs(self):
+        return zip(self.intergral_extension_psers, self.intergral_extension_pssers)
+
+    def test_from_to_pandas(self):
+        for pser, psser in self.intergral_extension_pser_psser_pairs:
             self.check_extension(pser, psser.to_pandas())
             self.check_extension(ps.from_pandas(pser), psser)
 
+    def test_isnull(self):
```
Comment (xinrong-databricks) [inline python/pyspark/pandas/tests/data_type_ops/test_num_ops.py:341]: Thanks for adding the tests!

File: python/pyspark/pandas/tests/data_type_ops/test_num_ops.py (hunk 2 above)
Comment (xinrong-databricks) [reply python/pyspark/pandas/tests/data_type_ops/test_num_ops.py:364]: Looks fine.

**Review by xinrong-meng — APPROVED** [review]

**Comment by ueshin** [comment]
Thanks! merging to master.

**Pull request closed without merging**